cpp/benchmarks/text/edit_distance.cpp

```diff
@@ -23,23 +26,100 @@ static void bench_edit_distance(nvbench::state& state)
     cudf::type_id::STRING, distribution_id::NORMAL, min_width, max_width);
   auto const strings_table = create_random_table(
     {cudf::type_id::STRING, cudf::type_id::STRING}, row_count{num_rows}, strings_profile);
-  cudf::strings_column_view input1(strings_table->view().column(0));
-  cudf::strings_column_view input2(strings_table->view().column(1));
+  auto input1 = strings_table->get_column(0);
+  auto input2 = strings_table->get_column(0);
+  auto sv1    = cudf::strings_column_view(input1.view());
+  auto sv2    = cudf::strings_column_view(input2.view());
 
   state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
 
-  auto chars_size =
-    input1.chars_size(cudf::get_default_stream()) + input2.chars_size(cudf::get_default_stream());
-  state.add_global_memory_reads<nvbench::int8_t>(chars_size);
+  state.add_global_memory_reads<nvbench::int8_t>(input1.alloc_size() + input2.alloc_size());
   // output are integers (one per row)
   state.add_global_memory_writes<nvbench::int32_t>(num_rows);
 
   state.exec(nvbench::exec_tag::sync,
-             [&](nvbench::launch& launch) { auto result = nvtext::edit_distance(input1, input2); });
+             [&](nvbench::launch&) { auto result = nvtext::edit_distance(sv1, sv2); });
+}
+
+static void bench_edit_distance_ascii(nvbench::state& state)
```

Comment:
This benchmark looks fine. Why is it so different from the the UTF-8 benchmark, though? Is it because the UTF-8 benchmark can use data generators for strings as-is, whereas for ASCII you have to do manual work? If that is indeed the case, then should we build that flexibility into the generators? `edit_distance` can't be the only API that needs that kind of flexibility (for instance, I'm assuming regex code paths would benefit too).